snuba/datasets/transactions.py

```diff
@@ -99,6 +100,33 @@ def __init__(self):
             timestamp_column='start_ts',
         )
 
+    def __update_options(self,
+        options: Optional[Mapping[str, Any]]=None,
+    ) -> Mapping[str, Any]:
+        if options is None:
+            options = {}
+        if "insert_allow_materialized_columns" not in options:
+            options["insert_allow_materialized_columns"] = 1
```

Comment:
Won't this not check correctly because options isn't mutable? I think you'll either need to copy this into a new dictionary on write (probably the best approach to avoid mutating the argument value) or make it a `MutableMapping` everywhere.

Reply:
good point. I did not check it again after adding the type hints. Will fix